### ISSUE TITLE
Scanning QR codes doesn't work when this addon is active

### ISSUE THREAD

**Issue by wehkah** [m-khvoinitsky/dark-background-light-text-extension#250]
When you scan a QR code image, you need to turn dark mode off for the current tab, otherwise the QR image will not be recognized. 

Examples:
- web.threema.ch: threema won't recognize the code until dark mode is turned off
- fritzbox showing wifi credentials: QR image is not recognized by QR scanner

I am not sure if this can be fixed in any other way than turning dark mode off temporarily, and if there is any way of making this addon recognize QR images. Therefore, as a possible solution, I suggest some kind of "quick action button" that can be put to Firefox's menu bar and/or to the context menu, that, when clicked upon, deactivates the addon for the current tab but only until the next reload of that page.

In other words: 
 - you click on this button, 
 - dark mode will be turned off, 
 - the page is shown in bright mode, 
 - you can now scan any code on that page,
 - as soon as the page reloads or the tab's content changes to the next page, dark mode will be reactivated.

Possible or not?

**Comment by m-khvoinitsky** [comment]
> Therefore, as a possible solution, I suggest some kind of "quick action button" that can be put to Firefox's menu bar and/or to the context menu, that, when clicked upon, deactivates the addon for the current tab but only until the next reload of that page.

Currently, you can press F2 (by default) to quickly disable the add-on for current tab and then F2 again to re-enable it.

In general this is happening because for QR code to have white outline. Some clever QR code generators know about it and generate background-agnostic QR codes which works good, others don't. As a workaround, it's possible to forcibly add white outlines to all QR codes like this:
```css
.qrcode {
  outline: 10px solid white;
}
```
The problem is, of cource, not all QR codes have class `qrcode` (this code will work on web.threema.ch but no idea about the others) so we have to collect a bunch of commonly used selectors.

I do not have fritzbox nearby, can you check which HTML-code is used for displaying QR code using Developer Tools and post here? I only need tag name, class and id.

**Comment by wehkah** [comment]
Thanks for the little piece of code, it works perfectly on Threema! :)

About the other information you asked, I am not sure which developer tools you meant, but I took a look at the pages' source code and these are the first few line of HTML source code that the Fritzbox generates:

```
<!DOCTYPE html>
<html lang="de">
<head>
<meta http-equiv=content-type content="text/html; charset=utf-8" />
<meta http-equiv="Cache-Control" content="private, no-transform" />
<meta http-equiv="X-UA-Compatible" content="IE=edge" />
<meta name="format-detection" content="telephone=no" />
<meta http-equiv="x-rim-auto-match" content="none" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes, minimal-ui" />
<meta name="mobile-web-app-capable" content="yes" />
<meta name="apple-mobile-web-app-capable" content="yes" />
<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />
<meta http-equiv="cleartype" content="on">
```

These header lines are the same on nearly every page. But when I go to the wifi page and call the data sheet, that contains the qrcode, a pop up window uses this source code below. I deleted everything that I believe not being related to qrcode. If you need anything more, just let me know:

```
<!DOCTYPE html>
<html>
<head>
<meta http-equiv=content-type content="text/html; charset=utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<title>WLAN-Zugang</title>
<link rel="stylesheet" type="text/css" href="/css/rd/print_old.css"/>
<link rel="stylesheet" type="text/css" href="/css/default/print.css"/>
<link rel="stylesheet" type="text/css" href="/css/default/printpreview.css" media="screen"/>
...
tr.qrcode_block td{
padding-top:1.25rem;
}
.qrcode_block span {
margin-left:0.625rem;
}
...
function init()
{
jsl.writeDataLabel("uiData");
if (true)
{updateQRCode("uiMainQRCode", "WIFI:S:_MYSSIDHERE_;T:WPA;P:_MYPASSWD_;;", 0.7);}
else
{
updateQRCode("uiMainQRCode", "WIFI:S:_MYSSID_;T:WPA;P:_MYPASSWD_;;", 1.5);
}
}
ready.onReadyPopup(init);
</script>
...
</body>
</html>
```

The actual qrcode image is generated by a Java script when the page is being delivered. I think this will make it a bit difficult to deal with. Using F2 is more then efficient on the Fritzbox, because it is not needed every day. But what about other sites that use the same technique or a similar one? If they also use a dynamically generated code, will there be any chance at all to make it work in dark mode?
